liquigraph-core/src/main/java/org/liquigraph/core/io/ReplaceChecksumGraphReader.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2014-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.liquigraph.core.io;
+
+import org.liquigraph.core.model.Changeset;
+
+import java.sql.Connection;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.toList;
+import static org.liquigraph.core.model.predicates.ChangesetChecksumIsEmpty.CHECKSUM_IS_EMPTY;
+
+public class ReplaceChecksumGraphReader implements ChangelogGraphReader {
```

Comment:
I don't think replacing checksums is related to read changelogs from a graph database (as `ChangelogGraphReader` suggests)
the operation of replacing checksums should be done when reading changesets in the original `ChangelogGraphReader` implementation, in my opinion

Reply:
Hi @fbiville,

Sorry, but I don't understand your suggestion.

ReplaceChecksumGraphReader is just a decorator delegating read functionnality to it delegate.
It gets called when MigrationRunner needs to read a persisted changelog:
MigrationRunner -> ReplaceChecksumGraphReader -> ChangelogGraphReaderImpl
and avoids to clutter ChangelogGraphReaderImpl with code related to updating checksums.

Do you prefer me to:

1. move the code from ReplaceChecksumGraphReader inside of ChangelogGraphReaderImpl ?
   i.e. something like calling a private method replaceEmptyChecksums that I define in ChangelogGraphReaderImpl from the ChangelogGraphReaderImpl#read ?
2. merge ReplaceChecksumGraphReader and ReplaceChecksumWriter classes and call this new class from ChangelogGraphReaderImpl#read method ? (similar to 1 in fact).
3. other ?

Thanks,
Adrian

Reply:
Hi, apologies for my unclear comment.
Indeed, I think option 1 would make more sense.

Reply:
done in commit 167e6de